src/solvers/dg/3d/dg.jl

```diff
@@ -1307,11 +1319,120 @@ function rhs!(dg::Dg3D, t_stage)
   @timeit timer() "source terms" calc_sources!(dg, dg.source_terms, t_stage)
 end
 
+# Calculate volume integral and update u_t
+apply_positivity_preserving_limiter!(dg::AbstractDg) = apply_positivity_preserving_limiter!(dg.elements.u, dg, equations(dg))
+
+# Only implemented and tested for compressible Euler equations in 3D
+function apply_positivity_preserving_limiter!(u, dg::AbstractDg, equation::AbstractEquation)
+end
+
+# Apply positivity limiter of Zhan and Shu to nodal values elements.u
+function apply_positivity_preserving_limiter!(u, dg::Dg3D, equation::CompressibleEulerEquations3D)
+  if dg.pp_limiter_apply
+    apply_positivity_preserving_limiter_density!(u, dg::Dg3D, equation::CompressibleEulerEquations3D)
+    apply_positivity_preserving_limiter_pressure!(u, dg::Dg3D, equation::CompressibleEulerEquations3D)
+  end
+end
+
+# Apply positivity limiter of Zhan and Shu to nodal values elements.u
+# density
+function apply_positivity_preserving_limiter_density!(u, dg::Dg3D, equation::CompressibleEulerEquations3D)
+  @unpack weights, pp_limiter_threshold = dg
+  Threads.@threads for element_id in 1:dg.n_elements
+    # Dermine minimum density
+    rho_min = 0.0
+    for k in 1:nnodes(dg), j in 1:nnodes(dg), i in 1:nnodes(dg)
+      u_node = get_node_vars(dg.elements.u, dg, i, j, k, element_id)
+      rho_min = min(rho_min, u_node[1])
+    end
+    U1_mean = 0.0
+    U2_mean = 0.0
+    U3_mean = 0.0
+    U4_mean = 0.0
+    U5_mean = 0.0
+    for k in 1:nnodes(dg), j in 1:nnodes(dg), i in 1:nnodes(dg)
+      u_node = get_node_vars(dg.elements.u, dg, i, j, k, element_id)
+      U1_mean += u_node[1] * weights[i] * weights[j] * weights[k] / 8.0
+      U2_mean += u_node[2] * weights[i] * weights[j] * weights[k] / 8.0 
+      U3_mean += u_node[3] * weights[i] * weights[j] * weights[k] / 8.0
+      U4_mean += u_node[4] * weights[i] * weights[j] * weights[k] / 8.0
+      U5_mean += u_node[5] * weights[i] * weights[j] * weights[k] / 8.0
+    end
+    # Detect if limiting is necessary
+    Theta_rho = 1
```

Comment:
Do they use a capital and a lowercase theta? If not, `theta` would be in agreement with the style guide.

Reply:
I second that!